### DISCUSSION TITLE
Repeat by Default option

### DISCUSSION THREAD

**Discussion by mcpalmer1980** [smplayer-dev/smplayer#135] [Ideas]
SMPlayer seems to tick the most boxes for me. I prefer how VLC has a one button AB loop instead of a 3 button popup, and how you can set the repeat mode globally, but it's useless because of the delay every time a new loop starts. SMPlayer has seamless looping and remembers AB loop positions and the repeat mode, but I want an option for new files to start on repeat mode. Ideally, a setting for short videos to start in repeat mode, perhaps with a slider to set the max length to be considered a short video. Yeah I know this sounds like a niche, almost pointless feature, but it's important to me.

I considered adding this feature myself but couldn't figure out how best to do so. I might try to find the code where EOF is detected and just force it to repeat in my own custom always repeating build, but a brief search has failed me thusfar. I have some C/C++ experience as a 90s teen game dev wannabe, but most of my experience is with python lately and I've don no UX work in C++.

**Comment by smplayer-dev** [comment]
Go to preferences -> advanced, type `repeat true` in the field to run actions whenever a file is opened.

**Comment by mcpalmer1980** [comment]
Never realized that I could do that.
Thanks
